File: leetcode/557. Reverse Words in a String III/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/reverse-words-in-a-string-iii/

const reverseWord = (word: string): string => {
  return word.split("").reverse().join("")
}

const reverseWords = (s: string) => {
  return s
    .split(" ")
    .map((word) => reverseWord(word))
    .join(" ")
}

console.log(reverseWords("Let's take LeetCode contest"))

File: leetcode/557. Reverse Words in a String III/main.go
package main

import "strings"

func reverseString(s string) string {
	runes := []rune(s)

	length := len(runes)

	for i := 0; i < length/2; i++ {
		runes[i], runes[length-i-1] = runes[length-i-1], runes[i]
	}

	reversedStr := string(runes)

	return reversedStr
}

func reverseWords(s string) string {
	words := strings.Fields(s)

	for i := 0; i < len(words); i++ {
		words[i] = reverseString(words[i])
	}

	return strings.Join(words, " ")

}
